codex/contracts/market.nim

```diff
@@ -123,19 +144,21 @@ method requestStorage(market: OnChainMarket, request: StorageRequest) {.async.}
 
 method getRequest*(
     market: OnChainMarket, id: RequestId
-): Future[?StorageRequest] {.async.} =
-  let key = $id
+): Future[?StorageRequest] {.async: (raises: []).} =
+  try:
+    let key = $id
 
-  if market.requestCache.contains(key):
-    return some market.requestCache[key]
+    if key in market.requestCache:
+      return some market.requestCache[key]
 
-  convertEthersError:
-    try:
-      let request = await market.contract.getRequest(id)
-      market.requestCache[key] = request
-      return some request
-    except Marketplace_UnknownRequest:
-      return none StorageRequest
+    let request = await market.contract.getRequest(id)
+    market.requestCache[key] = request
+    return some request
+  except Marketplace_UnknownRequest:
+    return none StorageRequest
+  except CatchableError as err:
```

Comment:
Because you removed `convertEthersError`, the exception type is likely `EthersError` here. By adding `except CatchableError`, we swallow `CancelledError`, which we don't want to do.